### PR TITLE
Updates and overhauls BoxStation

### PR DESCRIPTION
**Pull request by norsvenska**
This remakes and updates Boxstation to modern code, a project 7+ months in the making. I've tried to keep it as faithful to the original as possible (as such, most of the station layout is unchanged), while also updating it to fit in with other modern maps. Unless the admins have run it while I was offline, it has only been tested on live once, though I hope to get more tests in. Discord development thread is [here.](https://discord.com/channels/326822144233439242/1260278355161514034)

<details>
  <summary>Full SDMM map render</summary>
  
StrongDMM-2024-12-06 05 07 23

</details>

Some of the more major alterations:
<details>
  <summary>Aft medbay and virology</summary>
Virology used to be much, much bigger. The virology lab used to make up all of the non-virology rooms and halls in the screenshot (save for the breakroom and psych office, those were there. everything further down from the hangover spawn was virology), but not the virology pictured. Medical was missing content, and there was a massive, open maintenance area south of virology that was just begging to be developed.

StrongDMM-2024-12-06 05 10 14

Thus, virology was squashed and moved into that open maintenance area, a freezer and chemical storage closet were built, and the CMO's office was moved there. Speaking of...

StrongDMM-2024-12-06 05 15 07

Here's where the CMO's office used to be. I had tried to preserve it. Believe me. But, with the addition of the coroner, I just couldn't do it without making the coroner's office tiny. So, the CMO's old office was shrunk, the blast doors were removed, and the door was modified to only require medical access.

Other than that, Medical is fairly unchanged. Interestingly, it was the only department that had doubled wires. Those have been removed, of course.
</details>
<details>
  <summary>Ordnance, nanites, circuits, xenobiology. The science disciplines.</summary>
Let's start with the elephant in the room: nanites. Nanites do not exist anymore. What used to be the nanite lab has been repurposed into a circuits lab!

StrongDMM-2024-12-06 05 23 29

Now, in the one live test run, there was a scientist who could not find the departmental orders console (it's the one at the top of the above screenshot). It was rather funny, watching him try every other console in the department before he found it, but I worry that this might be a recurring theme. I, for one, think it's fine and easy to find, but I also placed it there. I'm not certain where I'd move it to if needed, but I'd be willing to do so. After all, I've dealt with worse over the past half a year.

On the topic of nanites, I did pay a little tribute. These are showcases; they aren't functional. But, you can find them in science maintenance.

StrongDMM-2024-12-06 06 07 14

Moving on to the ordnance lab.

StrongDMM-2024-12-06 05 41 24

If there was a competition for which room I modified most and came back to most often, this room would win. The 2020 toxins setup was much different from the 2024 ordnance setup. For reference of what I started with, here is a screenshot off of the Affectedarc webmap of Beestation's Boxstation.

image

There were a lot of things to change, I even had to move the launch room, bomb site, and that little airlock thingy between the launch room and lab four tiles east to make room for everything. And, I'm quite proud of how it turned out. I even wrote these two silly papers in the filing cabinet in the ordnance office.

image

Yes, the date on the left report is the IC date that Boxstation was removed. The right one is the date I started the development thread, as I did not keep the date I began working on it on record or anything. Not even on Github, since I only added it to my fork after I had spent a while working on it.

The xenobiology lab is mostly unchanged. What used to be an eleventh pen is now a cytology pen. There's also a little cytology lab setup near the entrance.

StrongDMM-2024-12-06 05 53 24

Genetics was one of the first rooms I remade. I'm not certain what it used to look like, due to the time passed, and none of the other Boxstations on the webmap have anything even close. I do remember having to reshape it because of the ordnance freezer chamber, but that's about it. Yes, the sink has been connected to the pipenet. It's actually been beneficial going room by room like this, otherwise I'd have missed the CytoPro and the sink. Anyway, that's the genetics lab. 

StrongDMM-2024-12-06 05 57 17

Much of the rest of this area is mostly unchanged. I did rearrange and redecorate the research lab, but that's really about it. Robotics is very untouched.

StrongDMM-2024-12-06 06 09 33

</details>

<details>
  <summary>Central Command, or in layman's terms, the bridge which happens to be in the center.</summary>
Most of the work here was just flooring. It was while working on this that I grew quite fond of the large wood floor tiles. The council chamber/meeting room was probably the room that I did the most to. It includes bridge assistant representation! He gets a little table with everything he needs to be a bridge assistant. I thought it was cute, I like it. The rest of the bridge is mostly the same. There are a couple of decals I added, but that's about it.

StrongDMM-2024-12-06 06 23 30

I did not touch the AI chamber for the most part. If you load up the old version of the map, it'll still be where it is now.

And, I suppose this is where I should talk about the other command areas.
Gateway and vault are pretty much the same. Just decal work for the most part. The vault safe has the hitman outfit, a lazarus, the reaper ID card, a stethoscope, a cash briefcase, and an Ansem/SC. All but the Ansem/SC can be found in other vaults, I chose the Ansem/FC over the more popular desert eagle for a weapon because of a) it's easy to get, you just need a black market uplink and a little cash, so it's not exactly exclusive loot; and b) it fits the vibe of maintenance reaper, a stealthy weapon that breaks lights.

EVA storage was much more of an undertaking. I vaguely remember it being separated into three very sparsely stocked rooms, and I didn't particularly like it. So, I tore all the walls down, put in some standard equipment, and turned the old command storage room (which contained, like, a signaler and that's it) into an EVA planning room for the more RP oriented space explorers (nobody). The idea was the RD would get up on "stage" (the wood floor) and explain what their mission was, and then they'd go do it. I was really hoping for a podium or something to be present, but alas.

StrongDMM-2024-12-06 06 27 08

</details>

<details>
  <summary>CARGONIA</summary>
Drone bay's there now. Bitrunning den was built into some maintenance. Boulder machine, Big Bess, and crates. Lots of crates. I will admit, I have never worked in a warehouse, but I tried to make the cargo bay somewhat realistic from a logistics standpoint, or at least, having the illusion of being realistic. The cargo warehouse, the little dirty one up top, has a dedicated express receiving area. Abandoned, of course. There's also a crate returns chute.

StrongDMM-2024-12-06 06 39 38

</details>

<details>
  <summary>Station Common Areas and the Service Department</summary>
Janitor closet.

image

Arrivals hallway, including the locker room, tool storage, and vacant office

image

The actual arrivals dock

image

Much of the work in the above areas was just floor decals. There's a barber shop in the locker room, including a razor which has been varedited to look like a hemostat, which just so happens to look similar to hair clippers, in a way. Arrivals security checkpoint was redone, too, into a customs post, which I think is nice.

Dormitories are cozy. It's lit by blacklights, which I think looks cool ingame. They have a laundry room, which only exists because otherwise it would be a five tile straight horizontal maintenance area, and there's a trophy case facing into the fore hallway with a couple flavorful items. From top to bottom, the Spinward Olympiad silver cup, a synthetic rose bouquet, the Spinward Olympiad gold cup, the NT-AEV 'Reach', and a historic L6 SAW (foam gun, don't panic. NT was being cheap.). It also has the showroom photo frames.

image

Fitness room, also mostly untouched. New weight lifting area above the holodeck, though.

image

Ah, the Grand Nanotrasen Diner. I went into designing this area with the goal of promoting cooperation as much as possible. In a perfect world, the clown and mime would be putting on a performance, the bartender would be serving drinks and food, an assistant the chef picked up off the street would be serving food, the chef would be making food, and the botanists would be growing food. Thus, the chef has access to the bar and hydroponics, the bartender has access to the kitchen, and so do the botanists. Backrooms are private, though. The botanists' trays are in terrariums to prevent the chef from doing their job for them. I know botanists never keep the normal setup, speaking as someone who has played a decent bit of botany, but it's the thought that counts. The service hall has been massively expanded. It used to just be a 3x5 room in maintenance, where the lathe is now, and I just didn't like that. Also, the fox in the wall is the Devourer of Gods. He drops a dark matter singularity toy when butchered. I thought it was funny.

StrongDMM-2024-12-06 07 07 45

The library, the chapel, and the departures lounge have also been changed. Because there was no art gallery, the chapel office was moved. Honestly, not sure what else to say on this.

StrongDMM-2024-12-06 07 14 13

</details>

<details>
  <summary>The Home of the Redshirts, Rival of the Greyshirts</summary>
Pretty much the whole area has been redone, I'm honestly not too sure where to begin with this. For one, the courtroom is now very colorful. The law office is pretty, the detective's office is pretty. I redid the interrogation room. The armoury has a formal uniform crate, like Metastation, and there are wardrobe closets for secoffs to play dress-up with. Mechbay, firing range. Perma has a full kitchen. If/when torment nexus gets merged, I'd add it to the space between the execution room and the interrogation room.

StrongDMM-2024-12-06 07 20 14

</details>

<details>
  <summary>Engi-nearing my FUCKING limit, or, I'm burning myself out having been writing this for the past 2.5 hours</summary>
Tech storage has new floors.

StrongDMM-2024-12-06 07 32 15

CE's office has been refloored and expanded by one tile. Foyer's mostly untouched, save for floor decals. Atmos work area retains the classic pipe dispensers.

StrongDMM-2024-12-06 07 33 17

Gravgen refloored, there's a shared engineering storage much like on Icebox, Supermatter engine is just the box setup.

StrongDMM-2024-12-06 07 35 33

Turbine engine.

StrongDMM-2024-12-06 07 37 35

Telecomms, which now has the capability to cool itself.

StrongDMM-2024-12-06 07 38 21

And then there's this little guy.

StrongDMM-2024-12-06 07 39 18

This is the atmospherics minisat. I needed a place for the HFR and crystallizer. Neither existed when Box was removed, and Icebox has the advantage of multi-z. So, I thought, I don't have anywhere to put it on station. I'll make my own station. And so I did. I've never operated the HFR, so I don't know if this is good enough room. Ideally, atmos techs would use space to add piping, but I digress. I think the atmos minisat is such a cool idea. I made an emergency shelter that people who fuck up would be able to flee to. They have control of the HFR room vents and scrubbers, and also, an emergency vent. Those blast doors open up to space. The idea was that if there was some massive fire or gas leak or something, they'd be able to get into the shelter, shut off the vents (so as to not waste air), and vent it until it's clear.
</details>

<details>
  <summary>Maintenance</summary>
There are a few rooms I added to maintenance. You may have already seen the abandoned nanite lab from the research section, here's the rest.

Old water utility room. It's flooded, and fishing in the water yields the same stuff from a toilet.

StrongDMM-2024-12-06 07 45 59

Old radiation shelter. Ever since maintenance has gotten radiation shielding, this shelter has stopped seeing usage. But, it remains as an emergency shelter. Comes with seven chairs, seven rations, and toys.

StrongDMM-2024-12-06 07 46 51

Homeless shelter. A little paper about stealing a cigarette machine to that the homeless give money back to the owner is on the table, right next to a brutal scene. Looks like the homeless caught on.

StrongDMM-2024-12-06 07 50 09

Mech bay and surgery den. Surgery den was untouched save for gib spawner and floors. Mech bay, I just thought it'd be cool. The fabricator isn't linked; it's the maint subtype.

StrongDMM-2024-12-06 07 51 27

Abandoned clinic. Contains medical loot, and is easily repairable.

StrongDMM-2024-12-06 07 53 21

Arnold's pizza shop. Contains Arnold pizza and the ingredients to make your own. 

StrongDMM-2024-12-06 07 56 00

Abandoned postal storage. Contains mailman hat, cheap lighter, package wrapping, junkmail, and destination tagger.

StrongDMM-2024-12-06 07 56 58

Abandoned EVA storage. Most of it's been picked clean, but there might still be some goodies scattered around. The fire axe cabinet is empty and broken.

StrongDMM-2024-12-06 07 59 11

</details>

Alright, that should cover most of the changes in the map. I've been typing this up for the past three hours, so my brain is no longer functioning at peak capacity. I enjoyed working on this over the past several months, and I hope it brings joy to those who play on it.

Please give feedback if you have any.

### CODE REVIEW

**Comment by norsvenska** [comment]
hmmmm
i suppose that's the problem with custom formatted papers. the map loads just fine, though.

looking to see if i can figure out a way that isn't 'delete the papers'

**Comment by EEASAS** [comment]
add a singular oxygen canjister to the ordnance lab please..

**Comment by norsvenska** [comment]
ordnance storage has two

image

**Comment by Kunstfreiheit** [comment]
nice PR

**Comment by Timonkeyn** [comment]
nice PR

**Comment by norsvenska** [comment]
tested a second time on round 244833
